src/grid/column.component.ts

```diff
@@ -0,0 +1,38 @@
+import { FilteringCondition } from "../main";
+import { DataType } from "../data-operations/data-util";
+import { AfterContentInit, Component, ContentChild, Input, TemplateRef } from "@angular/core";
+import { IgxCellHeaderTemplateDirective, IgxCellTemplateDirective } from "./grid.common";
+
+@Component({
+    moduleId: module.id,
+    selector: "igx-column",
+    template: ``
+})
+export class IgxColumnComponent implements AfterContentInit {
+
+    @Input() public field: string;
+    @Input() public header: string = "";
+    @Input() public sortable: boolean = false;
+    @Input() public editable: boolean = false;
+    @Input() public filtering: boolean = false;
+    @Input() public hidden: boolean = false;
+    @Input() public width: string;
+    @Input() public index: number;
+    @Input() public filteringCondition: Function = FilteringCondition.string.contains;
+    @Input() public filteringIgnoreCase: boolean = true;
+    @Input() public dataType: DataType = DataType.String;
+    @ContentChild(IgxCellTemplateDirective) protected cellTemplate: IgxCellTemplateDirective;
+    @ContentChild(IgxCellHeaderTemplateDirective) protected headTemplate: IgxCellHeaderTemplateDirective;
+
+    public bodyTemplate: TemplateRef<any>;
+    public headerTemplate: TemplateRef<any>;
```

Comment:
Should a `footerTemplate` also be added? Features like column summaries, for example, should be achieved with a footerTemplate IMO. What do you think?